Initial agent

diberry opened the pull request:
## Purpose
<!-- Describe the intention of the changes being proposed. What problem does it solve or functionality does it add? -->
* ...

## Does this introduce a breaking change?
<!-- Mark one with an "x". -->
```
[ ] Yes
[ ] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

<!-- Please check the one that applies to this PR using "x". -->
```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get the code

```
git clone [repo-address]
cd [repo-name]
git checkout [branch-name]
npm install
```

* Test the code
<!-- Add steps to run the tests suite and/or manually test -->
```
```

## What to Check
Verify that the following are valid
* ...

## Other Information
<!-- Add any other helpful information that may be needed here. -->

sinedied commented on packages/langgraph-agent/src/utils/update_env.ts:
```diff
@@ -0,0 +1,21 @@
+import fs from "fs";
```
you can directly use promises with:
```js
import fs from 'fs/promises';
```

sinedied commented on packages/langgraph-agent/src/scripts/loadVectorStore.ts:
```diff
@@ -0,0 +1,31 @@
+
+const NORTHWIND_PDF_DIRECTORY = "../data";
+
+async function loadData(embeddings: any): Promise<void> {
```
You can import `EmbeddingsInterface` from `"@langchain/core/embeddings"` instead of any

sinedied commented on packages/langgraph-agent/src/scripts/loadVectorStore.ts:
```diff
@@ -0,0 +1,31 @@
+  await loadData(embeddings);
+}
+
+main()
```
nit: it's usually best to avoid mixing promises and async/await, this can be refactored to a try/catch with a top-level await

sinedied reviewed (commented):
- nit: re file naming conventions, I see a mix of snake_case, camelCase and kebab-case. JS/TS typically use snake-case for file names, so I would suggest using that consistently. But maybe this was generated by LG studio?

- structure: 
  * Do you plan to add more packages to the project? If not, I think it's better to simplify the structure and move the langgraph-agent package to the root.
  * adding scripts to start the agent and open LG studio from the root would be a good idea (you can do it with `npm run xxx -ws <pkg>)

- Do you plan to add AZD support to generate the .env file? I think it would be a good idea to simplify the process of setting up the environment for new users. It can be added later though.

diberry commented:
@sinedied 

*  Moved to mostly snake_case. Did you mean JS uses mostly Kebab-case? I do realize that, I'm sticking with snake. I think there are several reasons to move to snake_case in general, regardless of language, especially when using Azure resources such as secrets in ACA. 

* structure: 
  * I added a very short fastify server in a separate package and a dockerfile to build both packages. 
  * I added a start script in the root package for the studio. I'm tracking a [Langstudio issue](https://github.com/langchain-ai/langgraph-studio/issues/258) to figure out how to resolve this for the devcontainer. 

* azd - I'm working on the infra. I would like to publish without it and circle back to it when it is complete. 

Based on these responses, can you approve the PR?
